python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -314,20 +315,29 @@ class NearestNeighbors(Base):
             When set to True, the kneighbors method will automatically
             convert the inputs to np.float32.
 
+        return_cupy : bool, optional (default = False)
+            When set to True, returns the outputs as cupy.ndarrays. This
+            prevents double conversion when using 'kneighbors_graph'.
+
         Returns
         -------
         distances: cuDF DataFrame or numpy ndarray
             The distances of the k-nearest neighbors for each column vector
             in X
 
-        indices: cuDF DataFrame of numpy ndarray
+        indices: cuDF DataFrame or numpy ndarray
             The indices of the k-nearest neighbors for each column vector in X
         """
 
         n_neighbors = self.n_neighbors if n_neighbors is None else n_neighbors
-        X = self.X_m if X is None else X
 
-        out_type = self._get_output_type(X)
+        use_training_data = False
```

Comment:
Might be a little more clean to set `use_training_data = X is None`, then it can be used below.